source/FreeRTOS_ARP.c

```diff
@@ -302,22 +302,32 @@ static TickType_t xLastGratuitousARPTime = 0U;
                                     eReturn = eReturnEthernetFrame;
                                 }
                             }
-                            else if( ulSenderProtocolAddress == ulTargetProtocolAddress ) /* Gratuitous ARP request? */
+                            else if( ( ulSenderProtocolAddress == ulTargetProtocolAddress ) &&
```

Comment:
Please add a comment to explain that the check is to ensure that the Gratuituous ARP packet is originated within the same subnet.